src/main/java/org/broadinstitute/consent/http/service/DatasetRegistrationService.java

```diff
@@ -407,17 +408,21 @@ private List<FileStorageObject> uploadFilesForDataset(
   }
 
   private List<FileStorageObject> uploadFilesForDatasetUpdate(
-      Map<String, FormDataBodyPart> files, Map<String, BlobId> uploadedFileCache, User user)
+      Map<String, FormDataBodyPart> files,
+      Map<String, BlobId> uploadedFileCache,
+      User user,
+      int idx)
       throws IOException {
     List<FileStorageObject> updateDatasetFSOs = new ArrayList<>();
-
-    if (files.containsKey(String.format(NIH_INSTITUTIONAL_CERTIFICATION_NAME, 0))) {
```

Comment:
This is the core of the bug.  When multiple datasets are updated, we can have multiple files submitted.  Assuming the `files` key was always `consentGroups[**0**].nihInstitutionalCertificationFile` is incorrect.  The update payload might be just a change for consentGroup[5], or a case where multiple consentGroup entries are updated at the same time.

Given this bug I've made [https://broadworkbench.atlassian.net/browse/DT-2759](https://broadworkbench.atlassian.net/browse/DT-2759)